docs/DESIGN.md

```diff
@@ -0,0 +1,54 @@
+# Design
+
+## Vision
+
+> High-level goals defining the long term vision for the project that will guide daily activities, its design, and feature acceptance.
+
+- **High performance**: auto-instrumentation performance impact should not be a concern for its users.
+- **Reliability**: stable and performant under different loads. Well-behaved under extreme load, with predictable, low resource consumption.
+- **Visibility**: users should be able to generate telemetry data that provides deep and detailed visibility into their applications. Such telemetry must allow users to identify and solve application-related issues in production.
+- **Useful out-of-the-box**: after install users should be able to get telemetry data from targeted libraries with none or minimal configuration (good selection of defaults).
+- **Extensible**: key components can be chosen either at build time.
```

Comment:
@open-telemetry/dotnet-maintainers TBH I do not understand what it means. It is copied from [Preliminary Roadmap Google doc](https://docs.google.com/document/d/1F25EzxYa7iSs2r9u0kjetCNPGS7Ui-bneHJEEwzEFR4/edit#heading=h.8ps4qge8rkv6)

Reply:
I think that the idea here is similar to #117 in that we want to allow configuration of what "components" to be used either at build time or runtime (through configuration). This would allow users to choose between enabling B3 or W3C (or both) context propagators for example. Or also allow a vendor to package and enable their plugins at build time.

Reply:
Please check if https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/137/commits/41ecd7a58f760542179f1dd52afc509d8c3321f4 is better. Feel free to directly commit changes :wink:

Reply:
That looks good to me!